### PR TITLE
Add exercises for seismic stations

### PR DESCRIPTION
**Pull request by seisman**
Closes https://github.com/seismo-learn/seismology101/issues/550.

### CODE REVIEW

**Review by seisman — COMMENTED** [review]
Fixes

**Review by seisman — COMMENTED** [review]
Fixes

File: source/exercises/station.md [hunk 1]
```diff
@@ -1 +1,232 @@
...
+## 使用 IRIS GMAP 筛选台站
+
+```{note}
+IRIS GMAP 使用了 Google 地图服务，因而需要科学上网才能正常访问。
+```
+
+IRIS GMAP 是一个由 IRIS 提供的地震台站网页搜索工具，可以方便地查看全球地震台站分布以及台站的
+详细信息。下面演示 IRIS GMAP 的基本使用方法。
+
+访问 IRIS GMAP 网站：[https://ds.iris.edu/gmap/](https://ds.iris.edu/gmap/)，
```
Comment (core-man) [inline source/exercises/station.md:34]:
```suggestion
访问 IRIS GMAP 网站 [https://ds.iris.edu/gmap/](https://ds.iris.edu/gmap/)，
```

File: source/exercises/station.md [hunk 2]
```diff
@@ -1 +1,232 @@
...
+会看到如下界面。界面左侧为功能栏，可以使用不同的准则筛选地震台站。
```
Comment (core-man) [inline source/exercises/station.md:35]:
```suggestion
会看到如下界面：界面左侧为功能栏，可以使用不同的准则筛选地震台站；
```

File: source/exercises/station.md [hunk 3]
```diff
@@ -1 +1,232 @@
...
+右侧为显示区，用于显示符合筛选条件的地震台站。
+
+左侧功能栏中，支持以多种不同的方式对地震台站进行筛选：
```
Comment (core-man) [inline source/exercises/station.md:38]: The list is a little long on the website. How about just using a sentence?
```
左侧功能栏中，支持以多种不同的方式对地震台站进行筛选：指定台网名、台站名、位置码、通道名；指定时间范围；指定台站位置范围（矩形区域或圆形区域）；指定地震数据中心。
```

File: source/exercises/station.md [hunk 4]
```diff
@@ -1 +1,232 @@
...
+:alt: "IRIS GMAP 界面"
+:width: 100%
+
+IRIS GMAP 界面
+:::
+
+现在我们要筛选出所有 `IU` 台网的宽频带地震台站，则可以在 **Network** 框中输入 `IU`，
+在 **Channel** 框中输入 `BH?`（此处的问号为通配符）。点击上方的 **Update Map**
+按钮，则界面右侧会显示出所有满足筛选条件的台站。右侧上方为地图区域，圆圈标记了
+台站的位置。下方为列表区域，会显示台网名、台站名、台站位置以及台站的开始和结束时间。
```
Comment (core-man) [inline source/exercises/station.md:56]:
```suggestion
例如，想筛选所有 `IU` 台网的宽频带地震台站，则可以在 **Network** 框中输入 `IU`，
在 **Channel** 框中输入 `BH?`（此处的问号为通配符），然后点击上方的 **Update Map**
按钮，界面右侧便会显示出所有满足筛选条件的台站。右侧上方为地图区域，圆圈标记了
台站的位置；下方为列表区域，会显示台网名、台站名、台站位置以及台站的开始和结束时间。
```

File: source/exercises/station.md [hunk 5]
```diff
@@ -1 +1,232 @@
...
+:::{figure} gmap-2.jpg
+:align: center
+:alt: "IRIS GMAP 显示 IU 台网的宽频带台站"
+:width: 100%
+
+IRIS GMAP 显示 IU 台网的宽频带台站
+:::
+
+更进一步，可以查看每个台站的详细信息。以位于美国新墨西哥州的地震台站 `IU.ANMO` 为例。
+点击 `IU.ANMO` 台站对应的圆圈，会出现如下图所示的提示框：
```
Comment (core-man) [inline source/exercises/station.md:67]:
```suggestion
可以更进一步查看每个台站的详细信息。以位于美国新墨西哥州的地震台站 `IU.ANMO` 为例，
点击 `IU.ANMO` 台站对应的圆圈或下方列表区域的该台，会出现如下图所示的提示框：
```

File: source/exercises/station.md [hunk 6]
```diff
@@ -1 +1,232 @@
...
+
+- 指定台网名、台站名、位置码、通道名
+- 指定时间范围
+- 指定台站位置范围（矩形区域或圆形区域）
+- 指定地震数据中心
+
+:::{figure} gmap-1.jpg
+:align: center
+:alt: "IRIS GMAP 界面"
+:width: 100%
```
Comment (core-man) [inline source/exercises/station.md:48]: I feel all the figures are a little large using 100%.

File: source/exercises/station.md [hunk 7]
```diff
@@ -1 +1,232 @@
...
+访问 IRIS GMAP 网站 [https://ds.iris.edu/gmap/](https://ds.iris.edu/gmap/)，
+会看到如下界面：界面左侧为功能栏，可以使用不同的准则筛选地震台站；
+右侧为显示区，用于显示符合筛选条件的地震台站。
+
+左侧功能栏中，支持以多种不同的方式对地震台站进行筛选：
+
+- 指定台网名、台站名、位置码、通道名
+- 指定时间范围
+- 指定台站位置范围（矩形区域或圆形区域）
+- 指定地震数据中心
```
Comment (seisman) [inline source/exercises/station.md:43]:
```suggestion
左侧功能栏中，支持以多种不同的方式对地震台站进行筛选：指定台网名、台站名、位置码、通道名；指定时间范围；指定台站位置范围（矩形区域或圆形区域）；指定地震数据中心。
```

File: source/exercises/station.md [hunk 8]
```diff
@@ -1 +1,232 @@
...
+
+:::{figure} gmap-1.jpg
+:align: center
+:alt: "IRIS GMAP 界面"
+:width: 100%
```
Comment (seisman) [inline source/exercises/station.md:48]:
```suggestion
:width: 90%
```

File: source/exercises/station.md [hunk 9]
```diff
@@ -1 +1,232 @@
...
+
+例如，想筛选所有 `IU` 台网的宽频带地震台站，则可以在 **Network** 框中输入 `IU`，
+在 **Channel** 框中输入 `BH?`（此处的问号为通配符），然后点击上方的 **Update Map**
+按钮，界面右侧便会显示出所有满足筛选条件的台站。右侧上方为地图区域，圆圈标记了
+台站的位置；下方为列表区域，会显示台网名、台站名、台站位置以及台站的开始和结束时间。
+
+:::{figure} gmap-2.jpg
+:align: center
+:alt: "IRIS GMAP 显示 IU 台网的宽频带台站"
+:width: 100%
```
Comment (seisman) [inline source/exercises/station.md:61]:
```suggestion
:width: 90%
```

File: source/exercises/station.md [hunk 10]
```diff
@@ -1 +1,232 @@
...
+IRIS GMAP 显示 IU 台网的宽频带台站
+:::
+
+可以更进一步查看每个台站的详细信息。以位于美国新墨西哥州的地震台站 `IU.ANMO` 为例，
+点击 `IU.ANMO` 台站对应的圆圈或下方列表区域的该台，会出现如下图所示的提示框：
+
+:::{figure} gmap-3.jpg
+:align: center
+:alt: "IRIS GMAP 查看 IU.ANMO 台站的基本信息"
+:width: 100%
```
Comment (seisman) [inline source/exercises/station.md:72]:
```suggestion
:width: 90%
```

File: source/exercises/station.md [hunk 11]
```diff
@@ -1 +1,232 @@
...
+表示台站依然在长期运行中）。
+
+点击提示框中的 “More Information”链接，则会跳转到 [IRIS MDA](https://ds.iris.edu/mda) 中
+该台站所对应的页面（即 https://ds.iris.edu/mda/IU/ANMO/?starttime=2002-11-19&endtime=2599-12-31)。
+该页面不仅列出了台站的基本信息，还列出了台站所使用的地震仪器及其基本参数。
+
+:::{figure} mda-1.jpg
+:align: center
+:alt: "IRIS MDA 中查看 IU.ANMO 台站的基本信息"
+:width: 100%
```
Comment (seisman) [inline source/exercises/station.md:87]:
```suggestion
:width: 90%
```

File: source/exercises/station.md [hunk 12]
```diff
@@ -1 +1,227 @@
...
+inv = read_inventory("stations.xml")
+```
+
+## 深入理解和使用 {class}`~obspy.core.inventory.inventory.Inventory` 类
+
+上面提到，{meth}`Client.get_stations() <obspy.clients.fdsn.client.Client.get_stations()>`
+和 {func}`read_inventory() <obspy.core.inventory.inventory.read_inventory>` 的返回值都是 
+{class}`~obspy.core.inventory.inventory.Inventory` 类型。
+事实上，{class}`~obspy.core.inventory.inventory.Inventory` 类是 ObsPy 中最核心的类之一，用于储存地震台站信息。
+下图展示了 {class}`~obspy.core.inventory.inventory.Inventory` 类的属性及其层级关系：
```
Comment (core-man) [inline source/exercises/station.md:165]:
```suggestion
下图展示了 {class}`~obspy.core.inventory.inventory.Inventory` 类的属性及其层级关系：
{class}`~obspy.core.inventory.inventory.Inventory` 类可以看做是 
{class}`~obspy.core.inventory.network.Network` 类的列表;
{class}`~obspy.core.inventory.network.Network` 类可以看做是 
{class}`~obspy.core.inventory.station.Station` 类的列表;
{class}`~obspy.core.inventory.station.Station` 类可以看做是 
{class}`~obspy.core.inventory.channel.Channel` 类的列表。
```

File: source/exercises/station.md [hunk 13]
```diff
@@ -1 +1,227 @@
...
+
+ObsPy 的 {class}`~obspy.core.inventory.inventory.Inventory` 类。引自 [ObsPy 网站](https://docs.obspy.org/_images/Inventory.png)。
+:::
+
+### {class}`~obspy.core.inventory.inventory.Inventory` 类
+
+{class}`~obspy.core.inventory.inventory.Inventory` 类可以看做是 
+{class}`~obspy.core.inventory.network.Network` 类的列表，因而可以对 
+{class}`~obspy.core.inventory.inventory.Inventory` 进行列表相关的操作。
+这里，我们对 `inv` 进行循环并打印每个元素（即 `Network` 类）的值：
```
Comment (core-man) [inline source/exercises/station.md:180]:
```suggestion
可以对 {class}`~obspy.core.inventory.inventory.Inventory` 进行列表相关的操作，
下面对 `inv` 进行循环并打印每个元素（即 `Network` 类）的值：
```

File: source/exercises/station.md [hunk 14]
```diff
@@ -1 +1,227 @@
...
+```{code-cell} ipython3
+for net in inv:
+    print(net)
+```
+
+### {class}`~obspy.core.inventory.network.Network` 类
+
+{class}`~obspy.core.inventory.network.Network` 类提供了很多台网相关的属性和函数。
+例如，下面的代码会输出台网代码、台网总台站数以及 `inv` 中包含的台站数目：
```
Comment (core-man) [inline source/exercises/station.md:189]:
```suggestion
例如，下面的代码会输出第一个台网的代码、总台站数以及 `inv` 中实际包含的台站数目：
```

File: source/exercises/station.md [hunk 15]
```diff
@@ -1 +1,227 @@
...
+```{code-cell} ipython3
+net = inv[0]
+print(net.code, net.total_number_of_stations, net.selected_number_of_stations)
+```
+
+{class}`~obspy.core.inventory.network.Network` 类可以看做是 
+{class}`~obspy.core.inventory.station.Station` 类的列表，因而可以对 
+{class}`~obspy.core.inventory.network.Network` 进行列表相关的操作。
+这里我们取其第一个台站并查看其信息：
```
Comment (core-man) [inline source/exercises/station.md:198]:
```suggestion
可以对 {class}`~obspy.core.inventory.network.Network` 进行列表相关的操作，
这里我们取其第一个台站并查看其信息：
```

File: source/exercises/station.md [hunk 16]
```diff
@@ -1 +1,227 @@
...
+{class}`~obspy.core.inventory.station.Station` 类也提供了很多台站相关的属性和函数。
+例如，下面的代码输出了当前台站的台站代码、经纬度、高程、台站的总通道数目和当前 `inv` 中包含的
+通道数目：
+```{code-cell} ipython3
+print(sta.code, sta.latitude, sta.longitude, sta.elevation)
+print(sta.total_number_of_channels, sta.selected_number_of_channels)
+```
+
+每个 {class}`~obspy.core.inventory.station.Station` 类可以看做是 
+{class}`~obspy.core.inventory.channel.Channel` 类的列表。这里我们取该台站的第一个通道并查看其信息：
```
Comment (core-man) [inline source/exercises/station.md:215]:
```suggestion
可以对 {class}`~obspy.core.inventory.station.Station` 进行列表相关的操作，
这里我们取该台站的第一个通道并查看其信息：
```

**Review by core-man — APPROVED** [review]

**Pull request merged**